test_util/cluster.py

```diff
@@ -0,0 +1,260 @@
+import itertools
+import logging
+import os
+import random
+import stat
+import string
+from collections import namedtuple
+from contextlib import closing, contextmanager
+from os.path import join
+
+import passlib.hash
+import pkg_resources
+from retrying import retry
+
+import test_util.ccm
+import test_util.installer_api_test
+import test_util.test_runner
+from ssh.ssh_tunnel import SSHTunnel, TunnelCollection
+
+
+Host = namedtuple('Host', ['public_ip', 'private_ip'])
+
+
+class TestCluster:
+
+    def __init__(
```

Comment:
I updated the interface -- the caller is now responsible for partitioning hosts by role and passing a callback for cleanup. There are then a couple convenience methods `from_vpc()` and `from_cloudformation()` that will take either a VPC or CF object and instantiate a `TestCluster` from them.